packages/rocket-uploads3-store-event-aws-infrastructure/src/s3-notification-event-store.ts

```diff
@@ -0,0 +1,68 @@
+import { Duration, RemovalPolicy, Stack } from '@aws-cdk/core'
+import { Bucket, EventType } from '@aws-cdk/aws-s3'
+import { RocketUtils } from '@boostercloud/framework-provider-aws-infrastructure'
+import { Function, Runtime, Code } from '@aws-cdk/aws-lambda'
+import { Table } from '@aws-cdk/aws-dynamodb'
+import { S3EventSource } from '@aws-cdk/aws-lambda-event-sources'
+import { createPolicyStatement } from '@boostercloud/framework-provider-aws-infrastructure/dist/infrastructure/stacks/policies'
+
+const path = require('path')
+
+export type AWSS3EventsLambdaParams = {
+  bucketName: string
+  eventTypeName: string
+  entityTypeName: string
+}
+
+export class S3NotificationEventStoreStack {
+  public static mountStack(params: AWSS3EventsLambdaParams, stack: Stack): void {
+    const appName = stack.stackName.split('-app')[0]
+
+    // Create new bucket to upload file
+    const sourceBucket = new Bucket(stack, 'sourceUploadBucket' + params.entityTypeName, {
+      bucketName: params.bucketName,
+      removalPolicy: RemovalPolicy.DESTROY,
+    })
+
+    const eventsStore = stack.node.tryFindChild('events-store') as Table
+
+    // Crete new Lambda function to be triggered once new file is uploaded in bucket
+    const processorFunction = new Function(stack, 'rocketS3Trigger' + params.entityTypeName, {
+      runtime: Runtime.NODEJS_12_X,
+      timeout: Duration.minutes(15),
+      memorySize: 1024,
+      handler: 'index.handler',
+      functionName: appName + '-' + params.entityTypeName.toLowerCase() +'-s3-rocket-trigger',
+      code: Code.fromAsset(path.join(__dirname, 'lambdas')),
```

Comment:
Does this work without zipping the lambda folder? Just saying because of the node_modules folder.

Reply:
yes, it works fine without zipping I tested this in a real app